### PR TITLE
Some fixes for doc

### PR DESCRIPTION
**Pull request by hkollmann**
- fix links and text in mobile
   - start doc of promise
   - fix server doc

### CODE REVIEW

File: docs/_sidebar.md [hunk 1]
```diff
@@ -157,3 +153,5 @@
     - [7. Translations](/tutorial/twitter/tutorial-part-7.md)
     - [8. Unit Testing](/tutorial/twitter/tutorial-part-8.md)
     - [9. Virtual Lists](/tutorial/twitter/tutorial-part-9.md)
+  - [Qooxdoo Mobile App](/mobile/tutorial.md)
```
Comment (derrell) [inline docs/_sidebar.md]: All of the items above this new one are numbered. Should not this one be too?

**Review by cboulanger — APPROVED** [review]
I did not even know we had setXxxAsync !!

File: docs/mobile/deployment.md [hunk 2]
```diff
@@ -14,24 +14,24 @@ On the Cordova website you find detailed tutorials which describe the deployment
 
 [Cordova Getting Started Tutorial](http://docs.phonegap.com/en/1.6.1/guide_getting-started_index.md.html#Getting%20Started%20Guides)
 
-Configure Cordova for deploying your %{Mobile} application
+Configure Cordova for deploying your Mobile application
 ----------------------------------------------------------
 
-After you installed cordova, and followed the introductions of [Cordova Getting Started Tutorial](http://docs.phonegap.com/en/1.6.1/guide_getting-started_index.md.html#Getting%20Started%20Guides), you are able to deploy your %{Mobile} application on your mobile device.
+After you installed cordova, and followed the introductions of [Cordova Getting Started Tutorial](http://docs.phonegap.com/en/1.6.1/guide_getting-started_index.md.html#Getting%20Started%20Guides), you are able to deploy your Mobile application on your mobile device.
```
Comment (derrell) [inline docs/mobile/deployment.md]: That documentation is seriously outdated...

Comment (derrell) [reply docs/mobile/deployment.md]: (Also old docs referenced elsewhere herein.)

**Review by derrell — CHANGES_REQUESTED** [review]
I don't know enough about Mobile to comment much on it, but I did have a few comments here...

File: docs/mobile/deployment.md (hunk 2 above)
Comment (derrell) [reply docs/mobile/deployment.md]: "That documentation" meaing the phonegap links

File: docs/_sidebar.md (hunk 1 above)
Comment (cboulanger) [reply docs/_sidebar.md]: I think it's a different tutorial specifically for mobile apps - the above ones belong to the website version.

**Comment by cboulanger** [comment]
I think the heading "Qooxdoo Tools" in the sidebar should really be called "Development" because taht is what this section is really about - how do I develop a qooxdoo application, using CLI, tools, and other specific information. 

To fix to fix https://github.com/qooxdoo/qooxdoo-compiler/issues/530 , we need 
 - to put an entry for ["Customising Application Startup"](http://www.qooxdoo.org/docs/#/compiler/CustomAppStartup)  in this section and
 - to put a minified URL of http://www.qooxdoo.org/docs/#/compiler/CustomAppStartup into https://github.com/qooxdoo/qooxdoo-compiler/blob/master/source/class/qx/tool/compiler/Console.js#L185

**Comment by cboulanger** [comment]
@hkollmann Can you intergrate this into your PR?

**Comment by hkollmann** [comment]
I am thinking about cleaning up the sidebar. IMHO it's to confusing because of it's length.
Any ideas how to cleanup the widget section?

**Comment by cboulanger** [comment]
Create a widget index/overview page containing the stuff that‘s currently in the sidebar....

**Comment by hkollmann** [comment]
During working on doc i found some more interesting topics in the old docs which we should rework, e.g. the stuff in documentation\manual\source\pages\data_binding. This is only partly converted to md format. 
I startet this work with renaming the folders to the old name so that it's easier to get an overwiew what's missing.
Help is welcomed!

**Comment by hkollmann** [comment]
to be reviewed:
   /pages/core  : missing documents
  /pages/data_binding: missing documents
  /pages/desktop: new Name gui with name changes.
 /pages/widgets: to be integrated
/pages/desktop to be integrated into /gui/README.md + adding missing docs.

+

**Review by cboulanger — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]
Thank you very much for this huge effort!!! 🥇

File: docs/core/oo_feature_summary.md [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+definition and it cannot take advantage of many crucial features of the
+build process (documentation, optimization, etc.).
+
+</div>
+
+    qx.Class.define("my.cool.Class",
+    {
+      statics:
+      {
+        driveLetter : "C"
```
Comment (derrell) [inline docs/core/oo_feature_summary.md]: Documenting what the defer section will do is very useful for later figuring out where the declarations of variables are located. I would encourage developers to add here:
```
drive : null  // set in defer method
```
 and a members section too:
```
members:
{
  whatsTheDrive : null  // set in defer method
}
```

File: docs/ebnf_like.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Syntax Diagrams
+
+A short summary of the elements used in syntax diagrams.
+
+| Syntax symbol     | Description                                               |
+| ----------------- | --------------------------------------------------------- |
+| **:=**            | production rule (*"non-terminal can be expanded to ..."*) |
+| **{** ... **}**   | 0..N occurrences (*"\*"*)                                 |
```
Comment (derrell) [inline docs/ebnf_like.md]: All of the EBNF documents above allow for a trailing comma even on the last item. We should at least document, here, that although they are used that way, the expectation is that the final element in a list will not have a trailing comma.

**Review by derrell — CHANGES_REQUESTED** [review]
I agree with @cboulanger. Herculean effort! Much appreciated.

I noticed a few things that should probably get fixed (or tell me it'll have to do, and I'll accept that)

**Review by oetiker — APPROVED** [review]

**Comment by hkollmann** [comment]
I just converted the old documentation and added some missing parts. I do not changed the content.
Could we make two PR of this: 
- This one with the old docs.
- A second one with content change?

@derrell : I would be very happy if you could do this.

There some more chapters in the old docs wich are worth to convert. I will do this in a later PR.

**Comment by derrell** [comment]
None of my comments are things that must go into this right now, so yes, I'll approve this PR and those changes can go into a future PR.

**Review by derrell — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]
wow, awesome amount of work!!  well done & thank you @hkollmann !!

**Review by level420 — APPROVED** [review]

**Pull request merged**